### PR TITLE
mgr/dashboard: bug fixes for rbd mirroring edit and promotion/demotion

### PR DESCRIPTION
**Pull request by Pegonzal**
Signed-off-by: Pedro Gonzalez Gomez <pegonzal@redhat.com>
fixes: https://tracker.ceph.com/issues/57922
This PR fixes a couple of bugs, both happening when using the rbd option mirroring journal.

One of them would happen when editing an image with the mirroring journal mode, the message "RBD [errno 22] invalid argument" would show up.

And the other would happen when using the promotion/demotion actions for journal mirroring.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [x] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-form/rbd-form.component.ts [hunk 1]
```diff
@@ -690,18 +690,15 @@ export class RbdFormComponent extends CdForm implements OnInit {
...
-      if (request.enable_mirror) {
-        request.features.push('journaling');
-      } else {
-        const index = request.features.indexOf('journaling', 0);
-        if (index > -1) {
-          request.features.splice(index, 1);
-        }
+      const index = request.features.indexOf('journaling', 0);
+      if (index > -1) {
+        request.features.splice(index, 1);
```
Comment (pereman2) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-form/rbd-form.component.ts:701]: Isn't the journaling feature required in pool mode?

Comment (Pegonzal) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-form/rbd-form.component.ts:701]: Yep, pushed the wrong changes. Fixed it already

Comment (Pegonzal) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/block/rbd-form/rbd-form.component.ts:701]: Changed it

**Review by nizamial09 — APPROVED** [review]
can this somehow tested using unit tests? or e2e?

**Comment by pereman2** [comment]
jenkins test dashboard

**Comment by pereman2** [comment]
jenkins test dashboard cephadm

**Review by pereman2 — APPROVED** [review]
LGETM, some tests would be nice

**Comment by Pegonzal** [comment]
jenkins test dashboard cephadm

**Pull request merged**